### ISSUE TITLE
TF-TRT model's TRTEngineOP loaded on the first GPU only

### ISSUE THREAD

**Issue by albe1022** [triton-inference-server/server#2002]
**Description**
When I load tf-trt resnet50 model on TRTIS with multiple GPUs, the server builds TRTEngineOPs during warm-up only on the first GPU, while model instances are loaded on all GPUs. Then, during inference, only the first GPU works while the others are kept idle.

Below is nvidia-smi output after loading is done:

```
Thu Sep 10 14:59:56 2020
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 440.33.01    Driver Version: 440.33.01    CUDA Version: 10.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Tesla T4            Off  | 00000000:14:00.0 Off |                    0 |
| N/A   47C    P0    27W /  70W |   9959MiB / 15109MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   1  Tesla T4            Off  | 00000000:15:00.0 Off |                    0 |
| N/A   39C    P0    27W /  70W |   1382MiB / 15109MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   2  Tesla T4            Off  | 00000000:39:00.0 Off |                    0 |
| N/A   41C    P0    26W /  70W |   1380MiB / 15109MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   3  Tesla T4            Off  | 00000000:3A:00.0 Off |                    0 |
| N/A   42C    P0    27W /  70W |   1380MiB / 15109MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   4  Tesla T4            Off  | 00000000:88:00.0 Off |                    0 |
| N/A   41C    P0    27W /  70W |   1380MiB / 15109MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   5  Tesla T4            Off  | 00000000:89:00.0 Off |                    0 |
| N/A   41C    P0    27W /  70W |   1380MiB / 15109MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   6  Tesla T4            Off  | 00000000:B1:00.0 Off |                    0 |
| N/A   41C    P0    27W /  70W |   1388MiB / 15109MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   7  Tesla T4            Off  | 00000000:B2:00.0 Off |                    0 |
| N/A   42C    P0    27W /  70W |   1380MiB / 15109MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0     89137      C   tritonserver                                9946MiB |
|    1     89137      C   tritonserver                                1369MiB |
|    2     89137      C   tritonserver                                1367MiB |
|    3     89137      C   tritonserver                                1367MiB |
|    4     89137      C   tritonserver                                1367MiB |
|    5     89137      C   tritonserver                                1367MiB |
|    6     89137      C   tritonserver                                1375MiB |
|    7     89137      C   tritonserver                                1367MiB |
+-----------------------------------------------------------------------------+
```

**Triton Information**
Triton Container 20.03.01

**To Reproduce**
load resnet50 model optimized through tf-trt on TRTIS with multiple GPUs

config.pbtxt

    name: "resnet"
    platform: "tensorflow_savedmodel"
    max_batch_size:100
    input [
      {
        name: "input_3"
        data_type: TYPE_FP32
        format: FORMAT_NHWC
        dims: [ 224,224,3 ]
      }
    ]
    output [
      {
        name: "probs"
        data_type: TYPE_FP32
        dims: [ 1000 ]
      }
    ]
    instance_group [
      {
        count: 1
        kind: KIND_GPU
        gpus:[0,1,2,3,4,5,6,7]
      }
    ]
    model_warmup [
      {
        name: "warmup"
        batch_size: 100
        inputs: {
            key: "input_3"
            value: {
                data_type: TYPE_FP32
                dims: [ 224,224, 3 ]
                random_data : True
            }
        }
      }
    ]

**Expected behavior**
TRTEngineOP is built on every GPU same as its model instance. 
All GPUs with model instances perform inference.

**Comment by GuanLuo** [comment]
Can you use a newer version of Triton (20.08 as for now) and check if the issue remains?

> Then, during inference, only the first GPU works while the others are kept idle.

Can you share how do you send out inference to Triton? Perhaps you are not generating enough load so that all inferences are completed by one GPU

**Comment by albe1022** [comment]
Same happened for 20.08 Triton.

I used perf_client_v2 with 20 concurrent requests.
As shown below, only the first gpu's utilization increased.


```
Thu Sep 17 10:59:24 2020
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 440.33.01    Driver Version: 440.33.01    CUDA Version: 10.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Tesla T4            Off  | 00000000:14:00.0 Off |                    0 |
| N/A   44C    P0    63W /  70W |   9877MiB / 15109MiB |     94%      Default |
+-------------------------------+----------------------+----------------------+
|   1  Tesla T4            Off  | 00000000:15:00.0 Off |                    0 |
| N/A   39C    P0    27W /  70W |    874MiB / 15109MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   2  Tesla T4            Off  | 00000000:39:00.0 Off |                    0 |
| N/A   41C    P0    26W /  70W |    878MiB / 15109MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   3  Tesla T4            Off  | 00000000:3A:00.0 Off |                    0 |
| N/A   41C    P0    27W /  70W |    886MiB / 15109MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   4  Tesla T4            Off  | 00000000:88:00.0 Off |                    0 |
| N/A   41C    P0    27W /  70W |    876MiB / 15109MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   5  Tesla T4            Off  | 00000000:89:00.0 Off |                    0 |
| N/A   41C    P0    27W /  70W |    876MiB / 15109MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   6  Tesla T4            Off  | 00000000:B1:00.0 Off |                    0 |
| N/A   41C    P0    27W /  70W |    876MiB / 15109MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   7  Tesla T4            Off  | 00000000:B2:00.0 Off |                    0 |
| N/A   41C    P0    27W /  70W |    874MiB / 15109MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0   2044855      C   tritonserver                                9864MiB |
|    1   2044855      C   tritonserver                                 861MiB |
|    2   2044855      C   tritonserver                                 865MiB |
|    3   2044855      C   tritonserver                                 873MiB |
|    4   2044855      C   tritonserver                                 863MiB |
|    5   2044855      C   tritonserver                                 863MiB |
|    6   2044855      C   tritonserver                                 863MiB |
|    7   2044855      C   tritonserver                                 861MiB |
+-----------------------------------------------------------------------------+

```

**Comment by GuanLuo** [comment]
@albe1022 Sorry for not getting back to you. Can you provide a minimal instructions to reproduce the issue? i.e. where to obtain the model, the setup of model repository and model config, etc.

**Comment by GuanLuo** [comment]
@albe1022 I can't reproduce the issue with current master. I use the TF-TRT model from this [post](https://developer.nvidia.com/blog/deploying-models-from-tensorflow-model-zoo-using-deepstream-and-triton-inference-server/) and the warm-up works fine and the load is distributed to different GPUs when running perf_client.

Can you try again or provide more detail on reproducing the issue? Otherwise I will close the ticket

**Comment by albe1022** [comment]
Thank you for your response.

I used following code to obtain the tf-trt optimized resnet50 model.

https://colab.research.google.com/github/vinhngx/tensorrt/blob/vinhn-tf20-notebook/tftrt/examples/image-classification/TFv2-TF-TRT-inference-from-Keras-saved-model.ipynb#scrollTo=qKSJ-oizkVQY

I think the differences from your model is that the model 1) is in savedmodel format and 2) is converted to INT8 precision.
The config file is provided at the first comment

**Comment by GuanLuo** [comment]
Yes, I am able to reproduce the issue by following the tutorial you linked, I will look into it. And I notice that the tutorial is using TF 2.0, I wonder if you have tried converting to TF-TRT using TF 1.x?

**Comment by albe1022** [comment]
No, I have tried TF 2.x only.
